api/api/admin/media_report.py

```diff
@@ -204,13 +212,106 @@ def queryset(self, request, qs):
                 qs = qs.order_by(
                     "-total_report_count", "-pending_report_count", "oldest_report_date"
                 )
+            if value is None:
+                qs = qs.filter(pending_report_count__gt=0)
+            if value == "prev":
+                qs = qs.filter(pending_report_count=0)
 
             return qs
 
     return PendingRecordCountFilter
 
 
-class MediaListAdmin(admin.ModelAdmin):
+class BulkModerationMixin:
+    def has_bulk_mod_permission(self, request):
+        return request.user.has_perm(f"api.add_{self.media_type}decision")
+
+    def _bulk_mod(self, request, queryset, action: DecisionAction):
+        """
+        Perform bulk moderation for the queryset as per the requested
+        action.
+
+        This follows the pattern of the ``delete_selected`` action
+        defined in ``django.contrib.admin.actions.py``.
+        """
+
+        opts = self.model._meta
+
+        if action == DecisionAction.MARKED_SENSITIVE:
+            initial_count = queryset.count()
+            queryset = queryset.filter(**{f"sensitive_{self.media_type}__isnull": True})
+
+            count = len(queryset)
+            stats = {
+                "selected count": initial_count,
+                "already sensitive": initial_count - count,
+                "count": count,
```

Comment:
```suggestion
                "count": count,
```

Nit: A more specific label here could be useful: "Sensitive count", "To mark sensitive", "New sensitive", as some options.

Reply:
That label was intentionally vague because this same view is used for all different bulk moderation actions. Updated to a more descriptive message in f8c0023.